### PR TITLE
Fix/Check to throw exception when domain is claimed by another organization

### PR DESCRIPTION
**Pull request by gbubemismith**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Added check to ensure an exception with the right message is thrown when another organization has claimed a domain.

## Code changes

* **src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs:** Added get claimed domain by domain name check and throw an exception if the list is not empty.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs [hunk 1]
```diff
@@ -39,6 +39,13 @@ public async Task<OrganizationDomain> VerifyOrganizationDomain(Guid id)
             throw new ConflictException("Domain has already been verified.");
         }
 
+        var claimedDomain =
+            await _organizationDomainRepository.GetClaimedDomainsByDomainNameAsync(domain.DomainName);
+        if (claimedDomain.Any())
+        {
+            throw new ConflictException("The domain has been verified by another organization.");
```
Comment (JaredSnider-Bitwarden) [inline src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs:46]: Does this verbiage match the verbiage on save that we return when a domain has already been verified?

Comment (gbubemismith) [reply src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs:46]: I changed it to what was reported on Jira. are you okay with the error message from the create endpoint?

Comment (JaredSnider-Bitwarden) [reply src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs:46]: I think using the same verbiage in both cases would be more consistent. 

On the create endpoint, we used: "The domain is not available to be claimed."  I'm going to condition off it to provide a translated message regardless so it's not that important that they match, but the consistency does add value, I think.

Comment (gbubemismith) [reply src/Core/OrganizationFeatures/OrganizationDomains/VerifyOrganizationDomainCommand.cs:46]: yeah that's alright

**Review by JaredSnider-Bitwarden — APPROVED** [review]
Looks great!

**Pull request merged**